rust/worker/src/blockstore/arrow/blockfile.rs

```diff
@@ -90,10 +99,43 @@ impl ArrowBlockfileWriter {
     ) -> Result<ArrowBlockfileFlusher, Box<dyn ChromaError>> {
         let mut delta_ids = HashSet::new();
         for delta in self.block_deltas.lock().values() {
+            // Skip empty blocks. Also, remove from sparse index.
+            if delta.len() == 0 {
+                tracing::info!("Delta with id {:?} is empty", delta.id);
+                self.sparse_index.remove_block(&delta.id);
+                continue;
+            }
             // TODO: might these error?
             self.block_manager.commit::<K, V>(delta);
             delta_ids.insert(delta.id);
         }
+        // We commit and flush an empty dummy block if the blockfile is empty.
+        // This is a bit weird but necessary. It can happen that
+        // other indexes of the segment are not empty. In this case,
+        // our segment open() logic breaks down since we only handle either
+        // all indexes initialized or none at all but not other combinations.
+        // We could argue that we should fix the readers to handle these cases
```

Comment:
I actually disagree this is weird and think and empty file should be allowed to exist. I can create an empty file on basically all filesystems I know.